File: 2026-08-15-csv-download-with-streaming/action-in-blog/src/main/java/action/in/blog/todo/domain/TodoEntity.java
package action.in.blog.todo.domain;

import jakarta.persistence.*;

import java.time.OffsetDateTime;

@Entity
@Table(name = "todos")
public class TodoEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    private String description;
    private boolean completed;
    @Column(name = "created_at", nullable = false)
    private OffsetDateTime createdAt;
    @Column(name = "updated_at", nullable = false)
    private OffsetDateTime updatedAt;

    public Long getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }

    public boolean isCompleted() {
        return completed;
    }

    public OffsetDateTime getCreatedAt() {
        return createdAt;
    }

    public OffsetDateTime getUpdatedAt() {
        return updatedAt;
    }

    @Override
    public String toString() {
        return "TodoEntity{" + "title='" + title + '\'' + '}';
    }
}

File: 2024-02-07-timezone-problem-of-jvm-application/action-in-blog/src/main/kotlin/blog/in/action/domain/TodoEntity.kt
package blog.`in`.action.domain

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "TB_TODO")
class TodoEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,
    val title: String = "",
    val content: String = "",
    val createdAt: LocalDateTime = LocalDateTime.now()
)